src/coreclr/vm/eetwain.cpp

```diff
@@ -1160,13 +1160,10 @@ HRESULT EECodeManager::FixContextForEnC(PCONTEXT         pCtx,
             if (pNewVar->startOffset <= newMethodOffset &&
                 pNewVar->endOffset   >  newMethodOffset)
             {
-                newMethodVarsSorted[(int)varNumber] = *pNewVar;
+                newMethodVarsSorted[varNumber] = *pNewVar;
             }
         }
 
-        _ASSERTE(newNumVars >= oldNumVars ||
```

Comment:
Had this capability been relaxed already?

Reply:
Unclear. The logic below seems to handle it and for whatever reason this has been there a long time so I assume was fixed long ago and has been firing for quite some time.

Reply:
I don't think this is handled properly, I think it can mean the JIT's frame layout changes and thus produce odd results after subsequent edits that add new locals. I think it is a Roslyn bug if it is removing locals in EnC edits.

Reply:
cc @tmat

Reply:
> I think it is a Roslyn bug if it is removing locals in EnC edits for active methods.

Thanks @jakobbotsch for keeping us honest. @tmat Is this something you are tracking? I can keep the assert in but it is firing a lot for the current suite of tests we are running.

Reply:
Agreed that I don't think ENC code is ever supposed to have fewer variables than the version before. Do we know for sure that Roslyn passed fewer variables or investigation is needed to determine if this is a runtime or Roslyn side issue?

Reply:
> Agreed that I don't think ENC code is ever supposed to have fewer variables than the version before.

That is indeed what all the docs say so with all this response I'm assuming this remains true.

> Do we know for sure that Roslyn passed fewer variables or investigation is needed to determine if this is a runtime or Roslyn side issue?

Unclear. My base assumption is that the metadata reading is correct and that the runtime is simply doing what it is told. This would mean Roslyn is producing a delta that has removed them. Let me try and root out where the value is coming from.

Reply:
Roslyn shouldn't remove locals. If you have a C# repro that does so, lmk.

Reply:
After adding some logging, here are the current details for the failing case:

```
TID 62f4: EECM::FixContextForEnC: Got old and new EnC preserved area sizes of 32 and 32
TID 62f4: EECM::FixContextForEnC: Old SP=00000072C51ED120, FP=00000072C51ED120
TID 62f4: EECM::FixContextForEnC: Old and new fixed stack sizes are 144 and 128
TID 62f4: EECM::FixContextForEnC: Old count: 5
TID 62f4: EECM::FixContextForEnC: Old var: 0
TID 62f4: EECM::FixContextForEnC: Old var: 1
TID 62f4: EECM::FixContextForEnC: Old var: 2
TID 62f4: EECM::FixContextForEnC: Old var: 3
TID 62f4: EECM::FixContextForEnC: Old var: 4
TID 62f4: EECM::FixContextForEnC: Old final count: 9
TID 62f4: EECM::FixContextForEnC: New count: 4
TID 62f4: EECM::FixContextForEnC: New var: 0
TID 62f4: EECM::FixContextForEnC: New var: 1
TID 62f4: EECM::FixContextForEnC: New var: 2
TID 62f4: EECM::FixContextForEnC: New var: 3
TID 62f4: EECM::FixContextForEnC: New final count: 8
TID 62f4: EECM::FixContextForEnC: new vars: 8 old vars: 9
```

Reply:
I'm going to revert the removal of this assert for now.